streaming/src/main/scala/org/apache/spark/streaming/receiver/BlockGenerator.scala

```diff
@@ -106,6 +106,7 @@ private[streaming] class BlockGenerator(
   private val blockQueueSize = conf.getInt("spark.streaming.blockQueueSize", 10)
   private val blocksForPushing = new ArrayBlockingQueue[Block](blockQueueSize)
   private val blockPushingThread = new Thread() { override def run() { keepPushingBlocks() } }
+  private val dataAddingLock = new Object
```

Comment:
Right, sorry, I thought of using it but I realized that it changes the locking logic significantly, which I dont want to do.